neard/src/runtime.rs

```diff
@@ -933,8 +936,12 @@ impl RuntimeAdapter for NightshadeRuntime {
     }
 
     fn get_epoch_protocol_version(&self, epoch_id: &EpochId) -> Result<ProtocolVersion, Error> {
-        let mut epoch_manager = self.epoch_manager.as_ref().write().expect(POISONED_LOCK_ERR);
-        Ok(epoch_manager.get_epoch_info(epoch_id)?.protocol_version)
+        if self.use_next_protocol_version {
+            Ok(PROTOCOL_VERSION + 1)
```

Comment:
This is most certainly not the desirable behavior. For the purposes for which a node might want to query the `epoch_protocol_version` it must always use the actual epoch protocol version on the chain.